File: app/src/main/java/com/fuckcoolapk/module/DisableBugly.java
package com.fuckcoolapk.module;

import android.content.Context;

import com.fuckcoolapk.utils.CoolapkContext;
import com.fuckcoolapk.utils.Log;
import com.fuckcoolapk.utils.OwnSP;

import de.robv.android.xposed.XC_MethodReplacement;
import de.robv.android.xposed.XposedBridge;
import de.robv.android.xposed.XposedHelpers;

public class DisableBugly {
    public void init() {
        if (OwnSP.INSTANCE.getOwnSP().getBoolean("disableBugly",false)){
            try {
                XposedBridge.hookAllMethods(XposedHelpers.findClass("com.tencent.bugly.crashreport.CrashReport", CoolapkContext.classLoader), "initCrashReport", new XC_MethodReplacement() {
                    @Override
                    protected Object replaceHookedMethod(MethodHookParam param) throws Throwable {
                        return null;
                    }
                });
            }catch (Throwable e){
                Log.e(e);
            }
        }
    }
}

File: app/src/main/java/com/fuckcoolapk/module/DisableBugly.kt
package com.fuckcoolapk.module

import com.fuckcoolapk.utils.CoolapkContext
import com.fuckcoolapk.utils.OwnSP
import com.fuckcoolapk.utils.ktx.replaceAfterAllMethods
import de.robv.android.xposed.XposedHelpers

class DisableBugly {
    fun init() {
        if (OwnSP.ownSP.getBoolean("disableBugly", false)) {
            XposedHelpers.findClass("com.tencent.bugly.crashreport.CrashReport", CoolapkContext.classLoader)
                    .replaceAfterAllMethods("initCrashReport") {
                        null
                    }
        }
    }
}